### PR TITLE
Implement Provider Reconciling for KubeVirt

### LINKED ISSUE #8144: Implement Provider Reconciling for KubeVirt

### PR DESCRIPTION
**Pull request by hdurand0710**
Signed-off-by: Helene Durand <helene@kubermatic.com>

**What does this PR do / Why do we need it**:
Implement the Provider Reconciling for Kubevirt for the CSI Access (Role and Rolebinding). As discussed, the needed ServiceAccount is not reconciled in this process, only created at cluster creation time.
This prepares also for the namespace creation in #8489 that will be reconciled too.

**Does this PR close any issues?**:
Fixes #8144

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/provider/cloud/kubevirt/provider.go [hunk 1]
```diff
@@ -84,6 +104,27 @@ func (k *kubevirt) ValidateCloudSpecUpdate(oldSpec kubermaticv1.CloudSpec, newSp
...
+		return nil, nil, err
+	}
+
+	client, restConfig, err := NewClientWithRestConfig(kubeconfig)
+	if err != nil {
+		return nil, nil, err
+	}
+	if err != nil {
+		return nil, nil, err
+	}
```
Comment (xrstf) [inline pkg/provider/cloud/kubevirt/provider.go]: Why is this here? ^^

Comment (hdurand0710) [reply pkg/provider/cloud/kubevirt/provider.go]: Thanks for noticing! Removed

File: pkg/provider/cloud/kubevirt/provider.go [hunk 2]
```diff
@@ -72,8 +74,26 @@ func (k *kubevirt) ValidateCloudSpec(spec kubermaticv1.CloudSpec) error {
 	return nil
 }
 
-func (k *kubevirt) InitializeCloudProvider(c *kubermaticv1.Cluster, p provider.ClusterUpdater) (*kubermaticv1.Cluster, error) {
-	return c, nil
+func (k *kubevirt) InitializeCloudProvider(cluster *kubermaticv1.Cluster, update provider.ClusterUpdater) (*kubermaticv1.Cluster, error) {
+	return k.reconcileCluster(cluster, update)
```
Comment (xrstf) [inline pkg/provider/cloud/kubevirt/provider.go:78]: Normally, the `InitializeCloudProvider` should be a no-op for cluster that have already been initialized. I can see that this is not talking to any cloud API, but I am still a bit worried that it might be too slow to reconcile KubeVirt clusters basically on every single cluster reconciliation.

Comment (xrstf) [reply pkg/provider/cloud/kubevirt/provider.go:78]: Nevermind, the InitializeCloudProvider isn't even used for providers that have a Reconcile function, so it doesn't matter.

**Review by xrstf — APPROVED** [review]
/approve
